components/tidb_query/src/rpn_expr/impl_encryption.rs

```diff
@@ -29,36 +33,106 @@ fn hex_digest(hashtype: MessageDigest, input: &[u8]) -> Result<Bytes> {
 
 #[cfg(test)]
 mod tests {
+    use tipb::ScalarFuncSig;
+
     use super::*;
     use crate::rpn_expr::types::test_util::RpnFnScalarEvaluator;
-    use tipb::ScalarFuncSig;
+
+    fn test_unary_func_ok_none<I: Evaluable, O: Evaluable>(sig: ScalarFuncSig)
+    where
+        O: PartialEq,
+        Option<I>: Into<ScalarValue>,
+        Option<O>: From<ScalarValue>,
+    {
+        assert_eq!(
+            None,
+            RpnFnScalarEvaluator::new()
+                .push_param(Option::<I>::None)
+                .evaluate::<O>(sig)
+                .unwrap()
+        );
+    }
 
     #[test]
-    fn test_sha1() {
-        let cases = vec![
-            (Some(""), Some("da39a3ee5e6b4b0d3255bfef95601890afd80709")),
-            (Some("a"), Some("86f7e437faa5a7fce15d1ddcb9eaeaea377667b8")),
-            (Some("ab"), Some("da23614e02469a0d7c7bd1bdab5c9c474b1904dc")),
+    fn test_md5() {
+        let test_cases = vec![
+            (vec![], "d41d8cd98f00b204e9800998ecf8427e"),
+            ("a".as_bytes().to_vec(), "0cc175b9c0f1b6a831c399e269772661"),
```

Comment:
I'm sorry, I missed the lint. You should use `b"a".to_vec()` for pure ascii str.